### ISSUE TITLE
"Unknown error. There was no error message to display." with blank Telemetry log lines

### ISSUE THREAD

**Issue by jrdioko** [rollbar/rollbar.js#560]
We recently started getting hundreds of occurrences a day of a Rollbar error that has the title, "Unknown error. There was no error message to display." and the following traceback:

```
Unknown error. There was no error message to display.
1
File "(unknown)" line (unknown) in [anonymous]
```

Looking into specific occurrences, the Telemetry data always contains one or more red error "Log" lines with a completely blank field on the right.

I was able to reproduce one of the occurrences, which was caused by this legacy code:

```
window.location.href = /sessions/new
```

The actual error displayed in the console in the browser was "Invalid regular expression flags" (with a jQuery backtrace), even though none of these details were included on the Rollbar side. Adding quotes around `/sessions/new` fixed that issue, but we are still getting hundreds of occurrences a day and have not been able to reproduce any errors in the browser.

What is causing this generic error message and blank Telemetry log lines? I have a support ticket open as well if engineers would like details on the specific Rollbar item (they requested I file a GitHub issue here).

**Comment by rivkahstandig3636** [comment]
Thanks for the bug report! We'll take a look into this.

**Comment by rokob** [comment]
I can reproduce the syntax error but I get the message logged properly to Rollbar. The message "Unknown error. There was no error message to display." happens when an exception is thrown that does not conform to the normal javascript error spec. This almost always is due to a specific browser creating an error object that is nonconforming. The specific problem here works in Firefox and Chrome, so I am going to make a guess that this happens on IE. Is the UA associated with the errors you are seeing all from IE? There isn't much we can do for these. Some browsers will print to their console properly but not produce a proper error object.

**Comment by jrdioko** [comment]
The UA on these errors is Chrome 63 and 64 (this is an internal tool, so the vast majority of users are on Chrome). The syntax error I mentioned was one example, but it looks like it was a small minority of the cases (we still see this happening around 2,000 times per day across various pages on the site).

I passed along the internal details to the support team (who then asked me to file an issue here). Is it possible for an engineer to look into that specific case and see what's going on?

**Comment by HugoHeneault** [comment]
I'm having this issue too. About 600 of them the last days... This is quite annoying... :(

**Comment by eyalkenig** [comment]
We are suffering from this as well. Started 2 months ago, with ~1500 occurrences until now.

**Comment by rivkahstandig3636** [comment]
This bug is scheduled to be worked on for the next release of rollbar.js. Sorry about this :(

**Comment by scott-ho** [comment]
@rivkahstandig3636 is this included in v2.4.0?

**Comment by rivkahstandig3636** [comment]
I don't believe so. @shakiba Would you be able to take this one on so we can get a fix out?

**Comment by Matrosskin** [comment]
Probably the issue is related to the issue in zone.js https://github.com/angular/zone.js/issues/1108 - it is already fixed in the master but it is not yet released. They pass into Rollbar's `onerror` handler the instance of `ErrorEvent` instead of arguments which usually passed by browsers (https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror). So as rollbar.js is not ready to process `ErrorEvent` object we have "Unknown error. There was no error message to display.". Also @rokob already mentioned:
>The message "Unknown error. There was no error message to display." happens when an exception is thrown that does not conform to the normal javascript error spec.

@jrdioko @HugoHeneault @eyalkenig can you please check and confirm if you use zone.js? Also I believe you have parameter `extraArgs` in payload for rollbar log with value `{isTrusted: true}`. Can you check it  also please?

**Comment by jrdioko** [comment]
Yes, we are using Angular, but I don't see `isTrusted: true` anywhere.

**Comment by Matrosskin** [comment]
@jrdioko do you have field `extraArgs` in the error logs at rollbar?

**Comment by jrdioko** [comment]
I'm afraid it's been so long since we ignored this error in Rollbar that it doesn't look like we have the historical errors available to look at anymore.

**Comment by scott-ho** [comment]
@Matrosskin There is no `extraArgs` in official rollbar docs nor in the typescript declaration file. For normal user, there is few chance to use it.

**Comment by Matrosskin** [comment]
@scott-ho sorry if I confuse you in my description of `extraArgs` - I did not mean that parameter as some argument for some method, so it should not be in the documentation or in typescript declaration. It is a property which can be sent as part of rollbar payload. So if it was sent then you will be able to see it only on the occurrence page. Is it possible if you post here the example of params (or raw json, probably it will be better) for one of your occurrences?

**Comment by Matrosskin** [comment]
While zone.js has not released fixed version we use next workaround: after initialization the Rollbar we override `window.onerror` to correct arguments:
```
const originOnError = window.onerror;
window.onerror = function(event) {
    if (originOnError) {
        // NOTE: In case error appear in script from separate domain then the event["filename"] will contain empty string.
        if (typeof event["filename"] != 'undefined') {
            return originOnError.call(window, event["message"], event["filename"], event["lineno"], event["colno"], event["error"]);
        } else {
            return originOnError.apply(window, arguments);
        }
    }
};
```
